amoro-common/src/main/java/org/apache/amoro/serialization/ResourceSerde.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.amoro.serialization;
+
+/**
+ * ResourceSerde interface
+ *
+ * @param <R> resource serde with obj
+ */
+public interface ResourceSerde<R> {
+
+  /**
+   * serialize resource
+   *
+   * @param resource input object
+   */
+  byte[] serializeResource(R resource);
+
+  /**
+   * deserialize resource
+   *
+   * @param input bytes
+   * @return output deserialize obj
+   */
+  DeserializedResource<R> deserializeResource(byte[] input);
+
+  final class DeserializedResource<R> {
+    private final R resource;
+    private final boolean modifiedDuringDeserialization;
+
+    public DeserializedResource(R resource, boolean modifiedDuringDeserialization) {
```

Comment:
what's the `modifiedDuringDeserialization` meanning?

Reply:
ModifiedDuringDeserialization when deserialization this is a behavior, such as t1 we have deposited in the object field. However, as time goes by, we may need to add t2 to the object, but the original object of the t2 field does not exist. We may have to perform some special assignment processing. At this point, we can determine whether additional processing is needed through this parameter.



Of course, for now we don't use it. If not needed, I can temporarily remove this parameter from this interface